1.1.6/flake.nix

```diff
@@ -0,0 +1,44 @@
+{
+  inputs = {
+    # Pinned to the commit that updated Terraform.
+    nixpkgs.url = "github:nixos/nixpkgs/3c2757d25d1ea5133fe8d207e995032b722b9e39";
+  };
```

Comment:
Think of https://github.com/NixOS/nixpkgs as a collection of packages akin to the entire Debian Stable APT repo. Except, it doesn't contain any binaries. Instead, nixpkgs contains all the source code needed to build the packages.

When you try to "install" a Nix package, first, Nix will see if it's already been built and exists in the binary cache. If not, Nix will build it from source.

Here is the package for Terraform: https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/networking/cluster/terraform/default.nix.

We are pinning our nixpkgs input to the commit that added Terraform 1.1.6. Another way of looking at it: we're operating off a snapshot of the operating system when they added Terraform 1.1.6. 

Now, we _could_ specify individual nixpkgs inputs for each package we're installing, but that may increase the image size and add additional complexity. We're fine with whatever version of the AWS CLI was present when they added Terraform 1.1.6.